Docker install fails on default packer script

the packer script fails as the way to install docker in amazon linux 2 has changed.

if you the script as it is now , you get the following error, i have put a pr #2 to fix this, also introduce nfs utils, if you want I can remove that PR and create a new one only with the change to the installer.

```
    amazon-ebs: Complete!
    amazon-ebs: Loaded plugins: priorities, update-motd
    amazon-ebs: No package docker available.
    amazon-ebs: Error: Nothing to do
==> amazon-ebs: Terminating the source AWS instance...
==> amazon-ebs: Cleaning up any extra volumes...
==> amazon-ebs: No volumes to clean up, skipping
==> amazon-ebs: Deleting temporary security group...
==> amazon-ebs: Deleting temporary keypair...
Build 'amazon-ebs' errored: Script exited with non-zero exit status: 1

==> Some builds didn't complete successfully and had errors:
--> amazon-ebs: Script exited with non-zero exit status: 1

==> Builds finished but no artifacts were created.
make: *** [ami] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -1,15 +1,39 @@
 KUBERNETES_VERSION ?= 1.10.3
+
 DATE ?= $(shell date +%Y-%m-%d)
-# Defaults to Amazon Linux 2 LTS Candidate AMI
-SOURCE_AMI_ID ?= ami-8c3848f4
+
+# Defaults to Amazon Linux 2 LTS AMI
+# * use the us-west-2 minimal hvm image
+# https://aws.amazon.com/amazon-linux-2/release-notes/
+SOURCE_AMI_ID ?= ami-37efa14f
+
+AWS_DEFAULT_REGION = us-west-2
 
 all: ami
 
-ami:
+validate:
+	packer validate eks-worker-al2.json
+
+ami: validate
 	packer build -var source_ami_id=$(SOURCE_AMI_ID) eks-worker-al2.json
 
 release:
-	aws s3 cp ./amazon-eks-nodegroup.yaml s3://amazon-eks/$(KUBERNETES_VERSION)/$(DATE)
+	AMI_ID=$(shell jq -r .builds[0].artifact_id manifest.json | cut -f2 -d':')
+	DESCRIPTION=$(shell aws ec2 describe-images --image-id $(AMI_ID) --query "Images[0].Description" --output text)
+	NAME=$(shell aws ec2 describe-images --image-id $(AMI_ID) --query "Images[0].Name" --output text)
+
+	# TODO: Get new image id, name, description
+	@echo "Copying image to us-east-1"
+	aws --region us-east-1 ec2 copy-image \
+		--source-region us-west-2 \
+		--source-image-id $(AMI_ID) \
+		--name $(AMI_NAME) \
+		--description $(DESCRIPTION) \
+		--query "ImageId" \
+		--output text
+	@echo "Sync the nodegroup yaml"
+	@echo "aws s3 cp ./amazon-eks-nodegroup.yaml s3://amazon-eks/$(KUBERNETES_VERSION)/$(DATE)"
+	@echo
 	@echo "Update CloudFormation link in docs at"
 	@echo " - https://github.com/awsdocs/amazon-eks-user-guide/blob/master/doc_source/launch-workers.md"
 	@echo " - https://github.com/awsdocs/amazon-eks-user-guide/blob/master/doc_source/getting-started.md"

--- a/eks-worker-al2.json
+++ b/eks-worker-al2.json
@@ -4,9 +4,10 @@
     "ami_name": "eks-k8s-worker-AL2-ami-{{timestamp}}",
     "binary_bucket_name": "amazon-eks",
     "binary_bucket_region": "us-west-2",
-    "binary_bucket_path": "1.10.3/2018-06-05/bin/linux/amd64",
+    "binary_bucket_path": "1.10.3/2018-07-26/bin/linux/amd64",
     "creator": "{{env `USER`}}",
-    "instance_type": "m4.large"
+    "instance_type": "m4.large",
+    "source_ami_id": ""
   },
 
   "builders": [
@@ -14,6 +15,17 @@
       "type": "amazon-ebs",
       "region": "{{user `aws_region`}}",
       "source_ami": "{{user `source_ami_id`}}",
+      "source_ami_filter": {
+        "filters": {
+          "architecture": "x86_64",
+          "name": "amzn2-ami-minimal-hvm-*",
+          "owner-id": "137112412989",
+          "root-device-type": "ebs",
+          "state": "available",
+          "virtualization-type": "hvm"
+        },
+        "most_recent": true
+      },
       "instance_type": "{{user `instance_type`}}",
       "launch_block_device_mappings": [
         {

--- a/files/kubelet-kubeconfig
+++ b/files/kubelet-kubeconfig
@@ -16,7 +16,7 @@ users:
   user:
     exec:
       apiVersion: client.authentication.k8s.io/v1alpha1
-      command: /usr/bin/heptio-authenticator-aws
+      command: /usr/bin/aws-iam-authenticator
       args:
         - "token"
         - "-i"

--- a/install-worker.sh
+++ b/install-worker.sh
@@ -17,6 +17,7 @@ sudo yum install -y \
     aws-cfn-bootstrap \
     conntrack \
     curl \
+    nfs-utils \
     socat \
     unzip \
     wget
@@ -44,7 +45,8 @@ sudo systemctl enable iptables-restore
 ################################################################################
 
 sudo yum install -y yum-utils device-mapper-persistent-data lvm2
-sudo yum install -y docker
+sudo amazon-linux-extras enable docker
+sudo yum install -y docker-17.06*
 sudo usermod -aG docker $USER
 
 # Enable docker daemon to start on boot.
@@ -84,12 +86,19 @@ if [ "$BINARY_BUCKET_REGION" = "us-east-1" ]; then
     S3_DOMAIN="s3"
 fi
 S3_URL_BASE="https://$S3_DOMAIN.amazonaws.com/$BINARY_BUCKET_NAME/$BINARY_BUCKET_PATH"
-wget $S3_URL_BASE/kubelet
-wget $S3_URL_BASE/kubectl
-wget $S3_URL_BASE/heptio-authenticator-aws
 
-chmod +x kubectl kubelet heptio-authenticator-aws
-sudo mv kubectl kubelet heptio-authenticator-aws /usr/bin/
+BINARIES=(
+    kubelet
+    kubectl
+    aws-iam-authenticator
+)
+for binary in ${BINARIES[*]} ; do
+    sudo wget $S3_URL_BASE/$binary
+    sudo wget $S3_URL_BASE/$binary.sha256
+    sudo sha256sum -c $binary.sha256
+    sudo chmod +x $binary
+    sudo mv $binary /usr/bin/
+done
 
 sudo mv $TEMPLATE_DIR/kubelet-kubeconfig /var/lib/kubelet/kubeconfig
 sudo mv $TEMPLATE_DIR/kubelet.service /etc/systemd/system/kubelet.service

